webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -741,11 +773,11 @@ class ComplianceAPIService(
 
     for {
       rules        <- getRules
-      allGroups    <- nodeGroupRepo.getAllNodeIds()
+      allGroups    <- nodeGroupRepo.getAllNodeIdsChunk()
       groupLib     <- nodeGroupRepo.getFullGroupLibrary()
       directiveLib <- directiveRepo.getFullDirectiveLibrary().map(_.allDirectives)
       allNodeInfos <- nodeInfoService.getAll()
-      nodeInfos    <- onlyNode match {
+      nodeTmpInfos <- onlyNode match {
                         case None     => allNodeInfos.succeed
                         case Some(id) =>
                           allNodeInfos
```

Comment:
Upmerge will be a little scary on this one because of the `NodeInfoService` to `NodeFactsRepo` migration, but the same implementation already exists in 8.1...

Apart from this block, there should be nothing that is non-complicated to change ^^